nnsvs/model.py

```diff
@@ -44,7 +43,7 @@ def __init__(self, in_dim, hidden_dim, out_dim, num_layers=4, dropout=0.0):
         ]
 
         self.model = nn.Sequential(*model)
-
+        self.prediction_type="deterministic"
```

Comment:
I am not a big fan of this approach... If we want to add a new class, then we need to write this line everytime :(

How about using the interface classes?  https://github.com/r9y9/nnsvs/blob/caefa1081bbd8d62ab9eb945033bd3ea02814d18/nnsvs/base.py.

I can think of two methods:

1. Prediction type as enum values (considering we have more types in the future)
```py
def prediction_type(self):
    return ${enum value is here} # defaults to deterministic?
```

2. Is deterministic or not

```py
def is_deterministic(self):
    return True
```


And then, inherit the base classes for all the model implementations, overwrite it if necessary. What do you think?

Reply:
I added enum type as PredictionType to base.py and predictin_type() to BaseModel, because we may have more types in the future.